### PR TITLE
fix: sidebar navigation on mobile

### PR DESCRIPTION
**Pull request by sshanzel**
## Changes
- We should always open the new sidebar on mobile.

https://github.com/dailydotdev/apps/assets/13744167/fb30eb79-6c4e-426c-8963-28d76b0069b4

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

MI-241 #done

### Preview domain
https://mi-241.preview.app.daily.dev

### CODE REVIEW

File: packages/webapp/components/layouts/AccountLayout/index.tsx [hunk 1]
```diff
@@ -44,10 +66,18 @@ export default function AccountLayout({
...
-          basePath="account"
-        />
+        {isMobile ? (
+          <ProfileSettingsMenu
+            isOpen={isOpen}
+            logout={logout}
+            onClose={closeSideNav}
+          />
+        ) : (
+          <SidebarNav
```
Comment (rebelchris) [inline packages/webapp/components/layouts/AccountLayout/index.tsx:76]: Ah was hoping we could remove it or uniform it.
Maybe check with design for a follow up task to uniform it? (non urgent)

File: packages/shared/src/components/drawers/NavDrawer.tsx [hunk 2]
```diff
@@ -64,7 +64,16 @@ export function NavDrawer({
...
+          <NavDrawerItem
+            key={item.label}
+            {...item}
+            onClick={(e) => {
+              if (item.onClick) {
+                item.onClick(e);
+              }
+              ref.current?.onClose();
+            }}
+          />
```
Comment (rebelchris) [inline packages/shared/src/components/drawers/NavDrawer.tsx:76]: Why is this needed?
Can't we uniform it inside the NavDrawerItem maybe?

File: packages/webapp/components/layouts/AccountLayout/index.tsx [hunk 3]
```diff
@@ -1,22 +1,44 @@
...
-import SidebarNav from './SidebarNav';
 import { getTemplatedTitle } from '../utils';
+import SidebarNav from './SidebarNav';
 
 export interface AccountLayoutProps {
   profile: PublicProfile;
   children?: ReactNode;
 }
 
+export const navigationKey = generateQueryKey(
```
Comment (rebelchris) [inline packages/webapp/components/layouts/AccountLayout/index.tsx:24]: Not urgent, but seeing it used in 2 places now would maybe make for a good hook?
I'm missing a bit of context here, why do we need react query for this?
Wouldn't simply state work?

**Review by rebelchris — COMMENTED** [review]
Minor comments, let me know what you think

File: packages/webapp/components/layouts/AccountLayout/index.tsx (hunk 1 above)
Comment (sshanzel) [reply packages/webapp/components/layouts/AccountLayout/index.tsx:76]: Yeah, at first I actually deleted this completely thinking we won't need it anymore, but then on laptop/tablet, they have a completely different design/components.

> Maybe check with design for a follow up task to uniform it? (non urgent)

Sure, will send a message.

File: packages/shared/src/components/drawers/NavDrawer.tsx (hunk 2 above)
Comment (sshanzel) [reply packages/shared/src/components/drawers/NavDrawer.tsx:76]: The ref is not found in the item itself. Unless we create a context or something to access this close function, we will have to make the call in this function.

File: packages/webapp/components/layouts/AccountLayout/index.tsx (hunk 3 above)
Comment (sshanzel) [reply packages/webapp/components/layouts/AccountLayout/index.tsx:24]: Because the back button is found not on where we initialized the sidebar.

File: packages/shared/src/components/drawers/NavDrawer.tsx (hunk 2 above)
Comment (rebelchris) [reply packages/shared/src/components/drawers/NavDrawer.tsx:76]: Hmm couldn't it just accept a ref as prop?
(anyway not breaking, more of a thought)

File: packages/webapp/components/layouts/AccountLayout/index.tsx (hunk 3 above)
Comment (rebelchris) [reply packages/webapp/components/layouts/AccountLayout/index.tsx:24]: Hmm seems like usePersistentContext almost?
Anyway up to you but dedicated hook might be needed just to maintain it better? (also optional)

Comment (sshanzel) [reply packages/webapp/components/layouts/AccountLayout/index.tsx:24]: Yeah, could've used the persistent context, though we don't need the extra usage of the idb storage. Maybe we should introduce a hook without it called `useQueryState`.

Comment (rebelchris) [reply packages/webapp/components/layouts/AccountLayout/index.tsx:24]: Hmm ah right, yes think in current flows we indeed just use dedicated hooks right to solve this.
We have quite a few that do something like this.

Comment (sshanzel) [reply packages/webapp/components/layouts/AccountLayout/index.tsx:23]: Let me know what you think of this https://github.com/dailydotdev/apps/pull/2867/commits/588741f91b43500a62ddb819c573be0f7b0707a1

File: packages/shared/src/components/drawers/NavDrawer.tsx (hunk 2 above)
Comment (sshanzel) [reply packages/shared/src/components/drawers/NavDrawer.tsx:76]: Yeah, we can do that. Clicking a nav item should automatically render a close action.

File: packages/shared/src/hooks/utils/useQueryState.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+type UseQueryState<T> = [
+  T,
+  (value: T) => void,
+  Omit<UseQueryResult<T>, 'data'>,
+];
+
+export const useQueryState = <T>(
+  key: QueryKey,
+  defaultValue: T,
+): UseQueryState<T> => {
```
Comment (rebelchris) [inline packages/shared/src/hooks/utils/useQueryState.ts:19]: Let's maybe make it object based (just in case we add more later)

**Review by rebelchris — APPROVED** [review]
Looks good to me.
Good hook to have in the belt.

@capJavert / @nensidosari thoughts?

File: packages/shared/src/hooks/utils/useQueryState.ts (hunk 4 above)
Comment (sshanzel) [reply packages/shared/src/hooks/utils/useQueryState.ts:19]: Yeah, this has no restriction, you can pass an object, boolean, string, or anything. Also we did not set a specific type.

Comment (rebelchris) [reply packages/shared/src/hooks/utils/useQueryState.ts:19]: I mean instead of plain props make it object props :) 

```
export const useQueryState = <T>({
  key
  defaultValue
}: PropsDef): UseQueryState<T> => {
```

Comment (sshanzel) [reply packages/shared/src/hooks/utils/useQueryState.ts:19]: Ahh, cool. You meant the props. I just thought of making it very similar to `useState`, but yeah, we can do that too.

Comment (sshanzel) [reply packages/shared/src/hooks/utils/useQueryState.ts]: I was just thinking we can have the optional parameters on the third when we need it. But yeah, updated the props now.

**Review by capJavert — APPROVED** [review]
I like the hook 👌
